cloudinit/net/networkd.py

```diff
@@ -546,69 +541,72 @@ def render_bonds(self, ns: NetworkState) -> dict:
+            cfg.update_section("NetDev", "Name", iface_name)
             cfg.update_section("NetDev", "Kind", "bond")
 
-            val = bond_cfg.get("mtu")
+            val = iface.get("mtu")
             if val:
                 cfg.update_section("NetDev", "MTUBytes", val)
 
-            val = bond_cfg.get("macaddress")
+            val = iface.get("mac_address")
             if val:
                 cfg.update_section("NetDev", "MACAddress", val.lower())
 
             # Optional bond parameters
-            params = bond_cfg.get("parameters", {})
-
             ParamMapType = Dict[str, Tuple[str, Callable[[object], str]]]
 
             param_map: ParamMapType = {
-                "ad-select": ("AdSelect", str),
-                "all-slaves-active": (
-                    "AllSlavesActive",
-                    lambda v: str(v).lower(),
-                ),
-                "arp-all-targets": ("ARPAllTargets", str),
-                "arp-interval": ("ARPIntervalSec", lambda v: f"{v}ms"),
-                "arp-validate": ("ARPValidate", str),
-                "down-delay": ("DownDelaySec", lambda v: f"{v}ms"),
-                "fail-over-mac-policy": ("FailOverMACPolicy", str),
-                "gratuitous-arp": ("GratuitousARP", str),
-                "lacp-rate": ("LACPTransmitRate", str),
-                "learn-packet-interval": ("LearnPacketIntervalSec", str),
-                "mii-monitor-interval": ("MIIMonitorSec", lambda v: f"{v}ms"),
-                "min-links": ("MinLinks", str),
-                "mode": ("Mode", str),
-                "packets-per-slave": ("PacketsPerSlave", str),
-                "primary-reselect-policy": ("PrimaryReselectPolicy", str),
-                "transmit-hash-policy": ("TransmitHashPolicy", str),
-                "up-delay": ("UpDelaySec", lambda v: f"{v}ms"),
+                "bond-ad-select": ("AdSelect", str),
+                "bond-arp-interval": ("ARPIntervalSec", lambda v: f"{v}ms"),
+                "bond-arp-validate": ("ARPValidate", str),
+                "bond-downdelay": ("DownDelaySec", lambda v: f"{v}ms"),
+                "bond-fail-over-mac": ("FailOverMACPolicy", str),
+                "bond-lacp-rate": ("LACPTransmitRate", str),
+                "bond-miimon": ("MIIMonitorSec", lambda v: f"{v}ms"),
+                "bond-min-links": ("MinLinks", str),
+                "bond-mode": ("Mode", str),
+                "bond-num-grat-arp": ("GratuitousARP", str),
+                "bond-primary-reselect": ("PrimaryReselectPolicy", str),
+                "bond-updelay": ("UpDelaySec", lambda v: f"{v}ms"),
+                "bond-xmit-hash-policy": ("TransmitHashPolicy", str),
+                # "all-slaves-active": ("AllSlavesActive", lambda v: str(v).lower()),
+                # "arp-all-targets": ("ARPAllTargets", str),
+                # "learn-packet-interval": ("LearnPacketIntervalSec", str),
+                # "packets-per-slave": ("PacketsPerSlave", str),
```

Comment:
I'm not sur how to handle these.

Reply:
I think this is another issue, and another PR. This is not a regression as it is currently not working either.

Reply:
https://github.com/canonical/cloud-init/pull/6555 should solve it